homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,254 @@
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+        key: str,
+        name: str,
+        icon: str,
+        device_class: str = None,
+        unit_of_measurement: str = None,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+        self._device_class = device_class
+        self._unit_of_measurement = unit_of_measurement
+
+        super().__init__(coordinator, location, device, key, name, icon)
+
+    @property
+    def device_class(self) -> str:
+        """Return the device class of the sensor."""
+        return self._device_class
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return self._unit_of_measurement
+
+
+class LyricIndoorTemperatureSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_indoor_temperature",
+            "Indoor Temperature",
+            "mdi:thermometer",
+            DEVICE_CLASS_TEMPERATURE,
+            hass.config.units.temperature_unit,
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self.device.indoorTemperature
+
+
+class LyricOutdoorTemperatureSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_temperature",
+            "Outdoor Temperature",
+            "mdi:thermometer",
+            DEVICE_CLASS_TEMPERATURE,
+            hass.config.units.temperature_unit,
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self.device.outdoorTemperature
+
+
+class LyricOutdoorHumiditySensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_humidity",
+            "Outdoor Humidity",
+            "mdi:water-percent",
```

Comment:
```suggestion
            None,
```


Don't set an icon if you set a device class.